src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -122,11 +137,42 @@ public MixpanelPushNotification(Context context, Notification.Builder builder, l
         mData.setMessage(message);
         mData.setLargeIconName(largeIconName);
         mData.setExpandableImageUrl(expandableImageURL);
-        mData.setBadgeCount(badgeCount);
         mData.setTag(notificationTag);
         mData.setGroupKey(groupKey);
         mData.setTicker(ticker);
         mData.setTimeString(timeString);
+        mData.setCampaignId(campaignId);
+        mData.setMessageId(messageId);
+        mData.setButtons(buildButtons(buttonsJsonStr));
+
+        int badgeCount = MixpanelNotificationData.NOT_SET;
+        if (null != badgeCountStr) {
+            try {
+
+                badgeCount = Integer.parseInt(badgeCountStr);
+                if (badgeCount < 0) {
+                    badgeCount = 0;
+                }
+            } catch(NumberFormatException e) {
+                badgeCount = 0;
+            }
+        }
+        mData.setBadgeCount(badgeCount);
+
+        int visibility = Notification.VISIBILITY_PRIVATE;
+        if (null != visibilityStr) {
+            switch (visibilityStr) {
+                case VISIBILITY_SECRET:
```

Comment:
For clarity, I'd prefer if we are more explicit. eg. `MixpanelPushNotification.VISIBILITY_SECRET`. Applicable to all VISIBILITY_*